### PR TITLE
Add create link button

### PR DESCRIPTION
**Pull request by gardenofwine**
p6_setup_create_link

screen shot 2015-03-03 at 18 03 26

### CODE REVIEW

File: app/src/main/java/com/paypal/psix/SetupEventFragment.java [hunk 1]
```diff
@@ -14,6 +18,40 @@
...
     @Override
     public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
-        return inflater.inflate(R.layout.fragment_setup_event, container);
+        View view = inflater.inflate(R.layout.fragment_setup_event, container);
+        addListeners(view);
+
+        return view;
+    }
+
+    private void addListeners(View rootView) {
```
Comment (ShayDavidson) [inline app/src/main/java/com/paypal/psix/SetupEventFragment.java]: You can get rid of this method by using Butterknife (merge from my branch which already has it or just add it to gradle if you don't want to merge (will only result in a gradle conflict).

Obtain the views using inject and add the listeners to them with butter knife as well. cleans up everything!

File: app/src/main/java/com/paypal/psix/SetupEventFragment.java [hunk 2]
```diff
@@ -14,6 +18,40 @@
...
+
+        final EditText paymentSumText = (EditText) rootView.findViewById(R.id.editPaymentSum);
+        paymentSumText.addTextChangedListener(toggleCreateLinkButtonTextWatcher);
+
+        final EditText paymentReasonText = (EditText) rootView.findViewById(R.id.editPaymentReason);
+        paymentReasonText.addTextChangedListener(toggleCreateLinkButtonTextWatcher);
+    }
+
+    private void toggleCreateLinkButton(){
+        Button createPaymentLink = (Button) getActivity().findViewById(R.id.buttonCreatePaymentLink);
```
Comment (ShayDavidson) [inline app/src/main/java/com/paypal/psix/SetupEventFragment.java]: obtain the button with butterknife.

File: app/src/main/java/com/paypal/psix/SetupEventFragment.java [hunk 3]
```diff
@@ -14,6 +18,40 @@
...
+            public void afterTextChanged(Editable s) {}
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {}
+
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                toggleCreateLinkButton();
+            }
+        };
+    }
+
+    private boolean isTextExist(int id){
```
Comment (ShayDavidson) [inline app/src/main/java/com/paypal/psix/SetupEventFragment.java]: doesTextExist

File: app/src/main/java/com/paypal/psix/SetupEventFragment.java [hunk 4]
```diff
@@ -14,6 +18,40 @@
...
+        final EditText paymentSum = (EditText) getActivity().findViewById(id);
```
Comment (ShayDavidson) [inline app/src/main/java/com/paypal/psix/SetupEventFragment.java]: butterknife

File: app/src/main/java/com/paypal/psix/SetupEventFragment.java (hunk 3 above)
Comment (ShayDavidson) [inline app/src/main/java/com/paypal/psix/SetupEventFragment.java]: or, `isDescriptionTextEmpty` for readability

File: app/src/main/res/layout/fragment_setup_event.xml [hunk 5]
```diff
@@ -1,47 +1,81 @@
...
+            android:layout_width="wrap_content"
             android:layout_height="wrap_content"
-            android:layout_weight=".5"
-            android:minHeight="50dp"
-            android:hint="@string/hint_setup_event_price"
-            android:inputType="number"
             android:textAppearance="?android:attr/textAppearanceLarge"
+            android:text="PH: Event Name"
+            android:layout_marginBottom="20dp"
+            android:id="@+id/eventName"
```
Comment (ShayDavidson) [inline app/src/main/res/layout/fragment_setup_event.xml]: event_name_text_view
i think ids look better in snake case rather than camel.

Comment (ShayDavidson) [inline app/src/main/res/layout/fragment_setup_event.xml]: don't forget these ids are global from the R object. `eventName` is too vague.

File: app/src/main/res/layout/fragment_setup_event.xml [hunk 6]
```diff
@@ -1,47 +1,81 @@
...
+            android:layout_gravity="center_horizontal" />
+
+        <TextView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:textAppearance="?android:attr/textAppearanceMedium"
+            android:text="@string/label_setup_event_price"
+            android:layout_marginBottom="30dp"
+            android:id="@+id/textView3"
```
Comment (ShayDavidson) [inline app/src/main/res/layout/fragment_setup_event.xml]: better id

File: app/src/main/res/layout/fragment_setup_event.xml [hunk 7]
```diff
@@ -1,47 +1,81 @@
...
+                android:ems="50"
+                android:id="@+id/editPaymentSum"/>
+        </LinearLayout>
+
+        <TextView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:textAppearance="?android:attr/textAppearanceMedium"
+            android:text="@string/label_setup_event_reason"
+            android:id="@+id/textView"
```
Comment (ShayDavidson) [inline app/src/main/res/layout/fragment_setup_event.xml]: better id

**Comment by ShayDavidson** [comment]
:dancer:

**Comment by avihut** [comment]
Apart from @ShayDavidson's comments, looks good :)

**Pull request merged**